File: sentimental-credit/credit.py
def main():
    card = get_card()
    sum = check_sum(card)
    card_type(card, sum)


def get_card():
    while True:
        card = input("Number: ")
        if card.isnumeric():
            break

    return card


def check_sum(card):
    sum = 0
    for i, digit in enumerate(reversed(card)):
        if i % 2 != 0:
            if int(digit) * 2 > 9:
                sum += int(digit) * 2 - 9
            else:
                sum += int(digit) * 2
        else:
            sum += int(digit)
    return sum % 10 == 0


def card_type(card, sum):
    if sum:
        if (int(card[0]) == 4) and (len(card) == 13 or len(card) == 16):
            print("VISA")
        elif (int(card[0:2]) == 34 or int(card[0:2]) == 37) and len(card) == 15:
            print("AMEX")
        elif int(card[0:2]) > 50 and int(card[0:2]) < 56:
            print("MASTERCARD")
        else:
            print("INVALID")
    else:
        print("INVALID")


main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long card;
    int sum = 0;
    int count = 0;

    do
    {
        card = get_long("Number: ");
    }
    while (card <= 0);

    // Get first 2 digits
    long firstTwoDigits = card;
    while (firstTwoDigits >= 100)
    {
        firstTwoDigits /= 10;
    }

    int firstDigit = firstTwoDigits / 10;

    // Validation Process
    while (card > 0)
    {
        int digit = card % 10;
        if (count % 2 == 1)
        {
            digit *= 2;
            while (digit > 0)
            {
                sum += digit % 10;
                digit /= 10;
            }
        }
        else
        {
            sum += digit;
        }
        card /= 10;
        count++;
    }

    // Print results
    if (sum % 10 == 0)
    {
        if ((firstTwoDigits == 34 || firstTwoDigits == 37) && count == 15)
        {
            printf("AMEX\n");
        }
        else if (firstDigit == 4 && (count == 13 || count == 16))
        {
            printf("VISA\n");
        }
        else if (firstTwoDigits > 50 && firstTwoDigits < 56)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}
